include/mutex.h

```diff
@@ -0,0 +1,37 @@
+#ifndef _MUTEX_
+#define _MUTEX_
+
+#include <atomic>
+
+//! Hybrid SpinMutex class
+//! \brief SpinMutex class that offers a fast locking and unlocking mechanism
+class SpinMutex {
+
+ public:
+  //! Attempt locking
+  bool try_lock() { return !lock_.test_and_set(std::memory_order_acquire); }
+
+  //! Call lock
+  void lock() {
+    std::size_t spin_count{0};
+    while (!try_lock()) {
```

Comment:
I wonder the double negative here, first in line 12 which returns `!` and then here. Is there any reason to do so?

Reply:
Because of the operation it is trying to do, we are trying to lock with the bool call. However, when we call the `try_lock` we are seeing if it can lock or not. We can't rename it as `try_unlock` because it won't be representative of what it means.

Reply:
Understood, thanks for the explanation